docs/api/openapi.yaml

```diff
@@ -0,0 +1,225 @@
+    For easy retrieval of a publication or data, assigned with a Dutch NBN in the Netherlands, a service called <a href
+    ='https://persistent-identifier.nl'>National Resolver</a> is available. This service is managed by <a href
+    ='https://dans.knaw.nl'>Data Archiving and Networked Services</a> (DANS) and <a href="https://www.kb.nl">KB, National Library of the Netherlands</a> (KB). Find out more about the <a href="https://www.kb.nl/organisatie/onderzoek-expertise/informatie-infrastructuur-diensten-voor-bibliotheken/registration-agency-nbn">'Registration Agency NBN'</a>.
+  license:
+    name: Apache License 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+  title: NBN:Reslover
+  version: 0.1.0
+
+externalDocs:
+
+  description: NBN Resolver on GitHub
+  url: 'https://github.com/DANS-KNAW/gmh-meresco/tree/brigmh'
+
+paths:
+
+  /nbn:
+    post:
+      summary: 'Registers a new URN:NBN identifier with associated locations.'
+      description: 'Registers a new URN:NBN associated with a prioritized list of locations. Multiple locations are prioritized in respective order. The first location is the preferred. The second is failover, etc. <br>The identifier must have a prefix that matches the authenticated user.'
+      operationId: 'createNbnLocations'
+      requestBody:
+        required: true
+        description: A json object that contains the URN:NBN and assosiated locations.
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/NbnLocationsObject'
+      responses:
+        '201':
+          description: Successful operation (created new)
+        '400':
+          description: Invalid URN:NBN identifier pattern or location uri(s) supplied
+        '401':
+            $ref: '#/components/responses/UnauthorizedError'
+        '403':
+          description: URN:NBN identifier is valid, but does not match the prefix of the authenticated user
+        '409':
+          description: Conflict, resource already exists
+            
+      tags:
+        - URN:NBN identifier
+
+  /nbn/{identifier}:
+    parameters:
+    - $ref: '#/components/parameters/nbn_identifier'
+    get:
+      summary: 'Returns URN:NBN object (urn:nbn+locations) for this {identifier}.'
+      description: 'Returns URN:NBN object (urn:nbn+locations) for this {identifier}.<br>The identifier must have a prefix that matches the authenticated user. HTTP 404 otherwise.'
+      operationId: 'getNbnRecord'
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/NbnLocationsObject'
+        '400':
+          description: Invalid URN:NBN identifier pattern supplied
+        '401':
+            $ref: '#/components/responses/UnauthorizedError'      
+        '404':
+          description: Supplied URN:NBN identifier not found, or URN:NBN-prefix is not registered to this user.
```

Comment:
"URN:NBN-prefix is not registered to this user" could also be a 403?

Reply:
Done